src/state/get-droppable-over.js

```diff
@@ -1,23 +1,60 @@
 // @flow
-import type { Position, Rect } from 'css-box-model';
+import type { Position, Rect, BoxModel } from 'css-box-model';
 import { toDroppableList } from './dimension-structures';
-import isPositionInFrame from './visibility/is-position-in-frame';
-import { find } from '../native-with-fallback';
+import { distance } from './position';
+import isWithin from './is-within';
+
 import type {
+  DroppableId,
   DroppableDimension,
+  DraggableDimension,
   DroppableDimensionMap,
-  DroppableId,
 } from '../types';
 
+type isCandidateArgs = {|
+  active: Rect,
+  target: Position,
+  client: BoxModel,
+|};
+
+const isCandidate = ({ active, target, client }: isCandidateArgs) => {
+  const isWithinVertical = isWithin(active.top, active.bottom)(target.y);
+  const isWithinHorizontal = isWithin(active.left, active.right)(target.x);
+  const { width: draggableWidth } = client.borderBox;
+  const activeCenterHorizontal = active.center.x;
+  const isCollidingLeft =
+    activeCenterHorizontal > target.x - draggableWidth / 2;
+  const isCollidingRight =
+    activeCenterHorizontal < target.x + draggableWidth / 2;
+
+  // TODO: shrink active
+
+  // TODO: current draggable rect have any overlap with smaller active
```

Comment:
This is where it's falling over currently. We need to figure out how to detect if the active `draggable` has any overlap with a `droppable`. Currently when we drag out of bounds (below the droppable's hit-zone) it will still be selected because it passes the first two edge detection cases. 

I feel compelled to check for an intersection but i'm wondering if there's a better way to do it. Also, we'll need to pass more information through as `active` since it's currently a center point (x, y), we will need `top`, `left`, `right`, `bottom` to calculate the intersection. Shouldn't be too hard to do if you think that's a good idea.

Reply:
We might be able to do-away with the edge-detection stuff below if we do that, come to think of it....